package/network/services/wireguard/Makefile

```diff
@@ -5,26 +5,43 @@
 #
 # This is free software, licensed under the GNU General Public License v2.
 # See /LICENSE for more information.
+#
+# When Linux 5.4 support is removed, kmod-wireguard should be moved to
+# package/kernel/linux/modules/netsupport.mk and this package removed.
 
 include $(TOPDIR)/rules.mk
 include $(INCLUDE_DIR)/kernel.mk
 
 PKG_NAME:=wireguard
 
-PKG_VERSION:=1.0.20201112
-PKG_RELEASE:=1
+PKG_CONFIG_DEPENDS:=CONFIG_LINUX_5_10
```

Comment:
This doesn't make sense.
Why not add it as a kernel module directly and rename the existing to wireguard-oot?

Reply:
there are dependencies in other modules on wireguard, including other repos (Luci), how would we manage that?

Reply:
Sorry for my poor Eng, I'll try to make it clear.

In `luci-proto-wireguard`, it depends on `kmod-wireguard` and `wireguard-tools`,
and we know, `wireguard-tools` actually depends on `kmod-wireguard` which is not added in it,
so just let `wireguard-tools` adjusts which package is needed, in-tree module or out-of-tree one,
like this: `+LINUX_5_4:kmod-wireguard-oot +!LINUX_5_4:kmod-wireguard`, then we can remove
`kmod-wireguard` in luci app.

Also, the oot one provided a meta package, see below:
https://github.com/openwrt/openwrt/blob/5bb9954826229782d4d285d65dfcca8a1c9e20d9/package/network/services/wireguard/Makefile#L50
https://github.com/openwrt/openwrt/blob/5bb9954826229782d4d285d65dfcca8a1c9e20d9/package/network/services/wireguard/Makefile#L53
We could reuse this as well, set `wireguard` as dependency in luci, and let it adjusts using the in-tree or oot,
but extra `VARIANT` is needed as build system will try to build the whole package by default.

Reply:
Okay I may do some work on this later today. @blocktrron I will let you know if a new commit is better for you to pick.

Reply:
https://github.com/openwrt/openwrt/pull/3885